File: bindings/php/tests/RandombytesTest.php
<?php

declare(strict_types=1);

use PHPUnit\Framework\TestCase;

/**
 * randombytes - no rejection/misuse category (a single `size` parameter, no key/tag to tamper
 * with or malform). Correctness: returns the requested length, and two calls are not identical.
 */
final class RandombytesTest extends TestCase
{
    public function testReturnsTheRequestedLength(): void
    {
        $this->assertSame(32, strlen(dstu_core_randombytes_buf(32)));
    }

    public function testReturnsEmptyForAZeroLength(): void
    {
        $this->assertSame('', dstu_core_randombytes_buf(0));
    }

    public function testDoesNotReturnTheSameBytesTwice(): void
    {
        $this->assertNotSame(dstu_core_randombytes_buf(32), dstu_core_randombytes_buf(32));
    }
}

File: bindings/java/src/test/java/ua/dstucrypto/dstucore/RandomBytesTest.java
package ua.dstucrypto.dstucore;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;

/**
 * {@code randombytes} - no rejection/misuse category (a single {@code int size} parameter, no
 * key/tag to tamper with or malform beyond what the type system already forecloses). Correctness:
 * returns the requested length, and two calls are not identical.
 */
class RandomBytesTest {
    @Test
    void returnsRequestedLength() {
        assertEquals(32, RandomBytes.buf(32).length);
    }

    @Test
    void zeroLengthReturnsEmpty() {
        assertEquals(0, RandomBytes.buf(0).length);
    }

    @Test
    void twoCallsAreNotIdentical() {
        assertFalse(java.util.Arrays.equals(RandomBytes.buf(32), RandomBytes.buf(32)));
    }
}
